bundle/src/main/java/com/adobe/acs/commons/wcm/impl/AemEnvironmentIndicatorFilter.java

```diff
@@ -193,10 +219,9 @@ public final void doFilter(final ServletRequest servletRequest, final ServletRes
                             ? capturedResponse.getBufferedServletOutput().getBufferedString()
                             : null;
 
-            
-                            
             if (contents != null
-                    && StringUtils.contains(response.getContentType(), "html")) {
+                    && StringUtils.contains(response.getContentType(), "html")
+                    && innerFilterAcceptsInjection(request)) {
```

Comment:
An alternative approach could be to remove the whole inner-filter registration and change this `&& innerFilterAcceptsInjection(request)` to `&& !isDisallowedWcmMode(WCMMode.fromRequest(request), excludedWcmModes)`, since the request at this point actually will have the `WCMMode` set, since it has gone through the `WCMRequestFilter`.

The only case where things could go wrong that I could think of, would be (custom) code that is changing the `WCMMode` to `DISABLED` and fails to change it back to the original value. This kind of construction is often in place to render certain components with the `DISABLED`-mode.